jas/common.go

```diff
@@ -62,33 +68,79 @@ func CreateJasScanner(serverDetails *config.ServerDetails, validateSecrets bool,
 			return
 		}
 	}
-	scanner = &JasScanner{}
-	if scanner.EnvVars, err = getJasEnvVars(serverDetails, validateSecrets, envVars); err != nil {
-		return scanner, err
-	}
+	// Create temp dir for scanner
 	var tempDir string
 	if tempDir, err = fileutils.CreateTempDir(); err != nil {
 		return
 	}
-	scanner.TempDir = tempDir
-	scanner.ScannerDirCleanupFunc = func() error {
-		return fileutils.RemoveTempDir(tempDir)
+	// Create scanner
+	scanner = &JasScanner{
+		ServerDetails: serverDetails,
+		TempDir:       tempDir,
+		ScannerDirCleanupFunc: func() error {
+			return fileutils.RemoveTempDir(tempDir)
+		},
+	}
+	// Apply options
+	for _, option := range options {
+		err = errors.Join(err, option(scanner))
 	}
-	scanner.ServerDetails = serverDetails
-	scanner.Exclusions = exclusions
-	scanner.MinSeverity = minSeverity
 	return
 }
 
-func getJasEnvVars(serverDetails *config.ServerDetails, validateSecrets bool, vars map[string]string) (map[string]string, error) {
+func WithEnvVars(validateSecrets bool, diffMode JasDiffScanEnvValue, envVars map[string]string) JasScannerOption {
+	return func(scanner *JasScanner) (err error) {
+		scanner.EnvVars, err = getJasEnvVars(scanner.ServerDetails, validateSecrets, diffMode, envVars)
+		return
+	}
+}
+
+func WithResultsToCompare(resultsToCompare *results.SecurityCommandResults) JasScannerOption {
+	return func(scanner *JasScanner) (err error) {
+		scanner.ResultsToCompare = resultsToCompare
+		return
+	}
+}
+
+func WithExclusions(exclusions ...string) JasScannerOption {
+	return func(scanner *JasScanner) (err error) {
+		scanner.Exclusions = exclusions
+		return
+	}
+}
+
+func WithMinSeverity(minSeverity severityutils.Severity) JasScannerOption {
+	return func(scanner *JasScanner) (err error) {
+		scanner.MinSeverity = minSeverity
+		return
+	}
+}
+
+func getJasEnvVars(serverDetails *config.ServerDetails, validateSecrets bool, diffMode JasDiffScanEnvValue, vars map[string]string) (map[string]string, error) {
 	amBasicVars, err := GetAnalyzerManagerEnvVariables(serverDetails)
 	if err != nil {
 		return nil, err
 	}
 	amBasicVars[JfSecretValidationEnvVariable] = strconv.FormatBool(validateSecrets)
+	if diffMode != NotDiffScanEnvValue {
```

Comment:
I think the if is redundant

Reply:
I only want to add this variable if its true.
If its false, its the same as not exists